libtock/alarm_timer.c

```diff
@@ -23,7 +23,7 @@ static void root_insert(alarm_t* alarm) {
   alarm_t **cur = &root;
   alarm_t *prev = NULL;
   while (*cur != NULL) {
-    if (cmp_exp(alarm->t0, alarm->expiration, (*cur)->expiration) < 0) {
+    if (cmp_exp(alarm->t0, alarm->interval, (*cur)->interval) < 0) {
```

Comment:
Should this be referenced from `alarm->t0` if comparing intervals now? I think this method of scheduling doesn't quite work anymore, this list ordering assumes a fixed time, but really we need to somehow understand what amount of an interval has expired when sorting the next one into the list.